bbl_screen-patch/kexec_ui/printerui/qml/Screen.qml

```diff
@@ -34,10 +34,15 @@ Rectangle {
     color: Colors.gray_500
     
     Component.onCompleted: {
-        if (X1PlusNative.getenv("KEXEC_LAUNCH_INSTALLER") != "")
-            dialogStack.push("SelectX1pPage.qml", { noBackButton: true } );
-        else
-            dialogStack.popupDialog("KexecDialog", {});
+        /* Boot flags set in check_kexec */
+        var KEXEC_LAUNCH_INSTALLER = X1PlusNative.getenv("KEXEC_LAUNCH_INSTALLER") || "";
+        var BOOT_DIALOG_TIMER = X1PlusNative.getenv("BOOT_DIALOG_TIMER") || 10;        
+        var X1P_OTA = X1PlusNative.getenv("X1P_OTA") || "";
+
+        if (KEXEC_LAUNCH_INSTALLER != "") //first stage installer is running, skip dialog
+            dialogStack.popupDialog("KexecDialog", { stage1: KEXEC_LAUNCH_INSTALLER, countdown: 0});
```

Comment:
if this skips the dialog why is it launching the dialog rather than just pushing SelectX1pPage?

Reply:
Good question. I will find a way to simplify this